src/wrapped_radix_engine_toolkit/low_level/compiled_notarized_intent.rs

```diff
@@ -0,0 +1,118 @@
+use crate::prelude::*;
+
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    Hash,
+    derive_more::Display,
+    derive_more::FromStr,
+    uniffi::Record,
+)]
+pub struct CompiledNotarizedIntent {
+    pub(crate) secret_magic: BagOfBytes,
+}
+
+impl CompiledNotarizedIntent {
+    pub fn new(bytes: BagOfBytes) -> Self {
+        Self {
+            secret_magic: bytes,
+        }
+    }
+
+    pub fn decompile(&self) -> Result<NotarizedTransaction> {
+        RET_decompile_notarize_tx(self.secret_magic.bytes())
```

Comment:
@0xOmarA would it be safe to unwrap here? or when can `decompile` fail **given a ` CompiledNotarizedIntent` which was created by RET (`compile` on `NotarizedTransactionV1`)**? Or when can it fail?